### ISSUE TITLE
Maximum image size?

### ISSUE THREAD

**Issue by Brubning** [dlemstra/Magick.NET#48]
I'm using Magick.Net for a project and getting the following exception:

{"Failed to allocate memory for custom tag binary object (150965188 elements of 1 bytes each). `C:\\..\\TestThumbs\\A2 YR11 WPP Grey.tif' @ error/tiff.c/TIFFErrors/572"}

The exception is thrown in the constructor (new MagickImage(filepath)). The image is 180MB, and while I'm building for Any CPU, prefer 32bit is set to true.

I'm using the 8-bit version of Magick.Net - could this be a colour space compatability problem? The TIF file is using 24bit colour depth.

Thanks

**Comment by dlemstra** [comment]
The file size of your image does not really matters. What matters are the dimensions of your image? You can use MagickImageInfo to get that information. A 32 bit system is limited around 2GB and I think you are hitting that memory limit. Are you able to read the file when you disable 'prefer 32 bit'?

**Comment by Brubning** [comment]
The image is 5250 x 2100 pixels, and the original format is TIFF. I've made a couple of changes to the way I was disposing the MagickImage object, I basically wasn't before and I think this was causing problems when I tried to use the byte array from the image. I can't really explain it, but I haven't seen the allocation error since the change.

Instead I started seeing a slightly changed issue; if I ran the following code I got a TIFF Error 572:

            using (var magickImg = new MagickImage(filepath) {Format = MagickFormat.Tiff})
            {
                ResizeImage(magickImg, maxSize);
                var bytes = magickImg.ToByteArray();   //Exception here with a TIFF file
                return bytes;
            }

I changed the Format to MagickFormat.Jpeg and it worked OK, same file. It bombs on the .ToByteArray() line.

I switched the build to x64, and all runs OK, so it does point to image size. The thumbnail it generates (which is 400x160 pixels) is 144MB, despite having much smaller dimensions. The image files (original and thumbnail) both claim to be uncompressed.

I've since put it back to Any CPU, but turned off Prefer 32bit and all is OK. I think I'll probably switch to dealing with these files as JPEGs, though!

Thanks
